src/yanfeishao/cs555/utils/OutputUtils.java

```diff
@@ -80,8 +81,67 @@ public void outputError(SimpleDBUtils simpleDBUtils, String prefix) {
+                    Date wifeDeathDate = familyEntity.getMother().getDeathDate();
+                    if (husbandBirthDate != null) {
+                        if (husbandBirthDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    if (husbandDeathDate != null) {
+                        if (husbandDeathDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    if (wifeBirthDate != null) {
+                        if (wifeBirthDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    if (wifeDeathDate != null) {
+                        if (wifeDeathDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    if (marriageDate != null) {
+                        if (marriageDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    if (divorceDate != null) {
+                        if (divorceDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    familyEntity.getChildList().forEach(child -> {
+                        try {
+                            Date childBirthDate = child.getBirthDate();
+                            Date childDeathDate = child.getDeathDate();
+                            if (childBirthDate != null) {
+                                if (childBirthDate.after(todayDate)) {
+                                    result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                                }
+                            }
+                            if (child.getDeathDate() != null) {
+                                if (childDeathDate.after(todayDate)) {
+                                    result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                                }
+                            }
+                        } catch (NullPointerException e) {
+                            // null birth/death date
```

Comment:
You do not need add try, catch logical, just add if (child != null) {....} at beginning this foreach loop